validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java

```diff
@@ -1331,6 +1320,14 @@ public void keyedValue(String nodeName, Object key, Object keyedValue) {
             @Nullable DefaultPropertyNode container) {
 
         final BeanIntrospection<Object> beanIntrospection = getBeanIntrospection(propertyValue);
+        AnnotationMetadata annotationMetadata = cascadeProperty.getAnnotationMetadata();
+        if (beanIntrospection == null && annotationMetadata.getDeclaredAnnotationNames().equals(
```

Comment:
`@Valid` is not annotated with `@Constraint` so you can just look for the presence of that annotation instead of comparing the annotation names